src/index.ts

```diff
@@ -16,12 +17,7 @@ const getOptionalInput = (name: string) => core.getInput(name) || undefined;
     return;
   }
 
-  const inputCwd = getOptionalInput("cwd");
-  if (inputCwd) {
-    core.info("changing directory to the one given as the input");
-    process.chdir(inputCwd);
-  }
-  const cwd = inputCwd || process.cwd();
+  const cwd = path.resolve(getOptionalInput("cwd") ?? "");
```

Comment:
@s0 I have simplified this slightly, this should still be equivalent, right? `path.resolve` already takes `process.cwd()` into account by default

Reply:
```
~ > node
> require('path').resolve('')
'/home/sam'
```

LGTM!